src/tests/verify_projects.py

```diff
@@ -55,36 +55,36 @@ def validate_project_file(self, project_filename: str) -> List[str]:
 
         errors = []
 
-        if not is_null(project, "ResourceAllocation", "XRAC"):
-            xrac = project["ResourceAllocation"]["XRAC"]
-
-            # Check 1
-            if not is_null(xrac, "ResourceGroups", "ResourceGroup"):
-                project_rgs = xrac["ResourceGroups"]["ResourceGroup"]
-                project_rg_names = [x["Name"] for x in project_rgs]
-
-                for rg in project_rg_names:
-                    if rg not in self.resource_group_names:
-                        errors.append(
-                            "%s: ResourceGroup '%s' not found in topology"
-                            % (project_filebn, rg)
-                        )
-
-            # Check 2
-            if not is_null(xrac, "AllowedSchedds", "AllowedSchedd"):
-                project_schedd_names = xrac["AllowedSchedds"]["AllowedSchedd"]
-                for sn in project_schedd_names:
-                    resource = self._get_resource_by_name(sn)
-                    if not resource:
-                        errors.append(
-                            "%s: AllowedSchedd '%s' not found in topology"
-                            % (project_filebn, sn)
-                        )
-                    elif "Submit Node" not in resource.service_names:
-                        errors.append(
-                            "%s: AllowedSchedd '%s' does not provide a Submit Node"
-                            % (project_filebn, sn)
-                        )
+        if not is_null(project, "ResourceAllocations", "ResourceAllocation"):
+            for resource_allocation in project["ResourceAllocations"]["ResourceAllocation"]:
+
+                # Check 1
+                if not is_null(resource_allocation, "ResourceGroups", "ResourceGroup"):
+                    project_rgs = resource_allocation["ResourceGroups"]["ResourceGroup"]
+                    project_rg_names = [x["GroupName"] for x in project_rgs]
+
+                    for rg in project_rg_names:
```

Comment:
Couldn't you speed this up with a set comparison?

Reply:
Sure can.